core/templates/dev/head/pages/exploration-player-page/services/learner-answer-info.service.spec.ts

```diff
@@ -296,42 +300,139 @@ describe('Learner answer info service', function() {
       'LearnerAnswerDetailsBackendApiService');
     DEFAULT_OUTCOME_CLASSIFICATION = $injector.get(
       'DEFAULT_OUTCOME_CLASSIFICATION');
-    state = sof.createFromBackendDict('new state', stateDict);
+    firstState = sof.createFromBackendDict('new state', stateDict);
+    secondState = sof.createFromBackendDict('fake state', stateDict);
+    thirdState = sof.createFromBackendDict('demo state', stateDict);
     spyOn(
       mockAnswerClassificationService,
       'getMatchingClassificationResult').and.returnValue(acrof.createNew(
       oof.createNew('default', 'default_outcome', '', []), 2, 0,
       DEFAULT_OUTCOME_CLASSIFICATION));
+    mockAnswer = 'This is my answer';
+    mockInteractionRulesService = 'Sample interaction rules service';
```

Comment:
This should be a service, not a string.

Why not just use one of the actual interaction rules services?